### PR TITLE
Deployment test

### PR DESCRIPTION
**Pull request by dgavrischuk**
Deployment test added
Folders structure changed

### CODE REVIEW

File: Google/resource-manager/folder/folders.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+project_id: "decisive-coda-355402"
```
Comment (AlexBMes) [inline Google/resource-manager/folder/folders.yaml]: I don't think this file should be here. If it is here as an example of the structure, I think it should be present in a README. 
Also might be better to use project and folder IDs that don't actually exist i.e just use `project-id` and `folders/1234567`

File: .github/workflows/deployment_tests.yml [hunk 2]
```diff
@@ -39,199 +37,13 @@ jobs:
...
-      # Install kubectl and create gke cluster
-      - name: setup kubectl
-        run: |
-          gcloud components install kubectl
-          gcloud container clusters create-auto mcpdeploytest-cluster --region=europe-west2
-          gcloud container clusters get-credentials mcpdeploytest-cluster --region=europe-west2
-
-      # Setup main.tf, deploy and destroy k8s
-      - name: deploy k8s
+      - name: Deploy Google folder
```
Comment (AlexBMes) [inline .github/workflows/deployment_tests.yml:48]: Should add a step before this to configure the files with project and folder IDs retrieved from github secrets. (see [mcp-services deployment tests](https://github.com/mesoform/Multi-Cloud-Platform-Services/blob/main/.github/workflows/deployment_tests.yml#L74-L80))

File: tests/gcp/deployment/google_folder/locals.tf [hunk 3]
```diff
@@ -0,0 +1,10 @@
+locals {
```
Comment (AlexBMes) [inline tests/gcp/deployment/google_folder/locals.tf]: Should be deleted if using module structure described in main review comment

File: tests/gcp/deployment/google_folder/main.tf [hunk 4]
```diff
@@ -0,0 +1,41 @@
+locals {
```
Comment (AlexBMes) [inline tests/gcp/deployment/google_folder/main.tf]: Should be replaced with
```hcl
module folders {
  source = "../../../../Google/resource-manager/folder/main.tf"
}
```
As stated in main review comment

File: tests/gcp/deployment/google_folder/provider.tf [hunk 5]
```diff
@@ -0,0 +1,15 @@
+terraform {
```
Comment (AlexBMes) [inline tests/gcp/deployment/google_folder/provider.tf]: Should also be deleted as using module block to reference

File: tests/gcp/deployment/google_folder/resource/folders.yaml [hunk 6]
```diff
@@ -0,0 +1,31 @@
+project_id: "decisive-coda-355402"
```
Comment (AlexBMes) [inline tests/gcp/deployment/google_folder/resource/folders.yaml]: Should be replaced with a string for replacement using `sed`. See [cloudrun.yaml](https://github.com/mesoform/Multi-Cloud-Platform-Services/blob/main/tests/mcp/deployment/GCP/gcp_cloudrun.yml#L1) in mcp-services repo, which gets replaced in [deployment tests step](https://github.com/mesoform/Multi-Cloud-Platform-Services/blob/main/.github/workflows/deployment_tests.yml#L78)

File: tests/gcp/deployment/google_folder/resource/folders.yaml [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+location_id: "europe-west1"
+create_google_project: "false"
+components:
+  parent_id: "folders/227414018885"
```
Comment (AlexBMes) [inline tests/gcp/deployment/google_folder/resource/folders.yaml]: This should also be replaced by a folder ID retrieved from a secret in the deployment test.  
@gbmeuk would the secret be called `GCP_FOLDER_ID`?

File: tests/gcp/deployment/google_folder/vars.tf [hunk 8]
```diff
@@ -0,0 +1,3 @@
+variable "folders_yml" {
```
Comment (AlexBMes) [inline tests/gcp/deployment/google_folder/vars.tf]: Should also be deleted as using module block to reference

File: tests/gcp/unit_tests/goole_folder/locals.tf [hunk 9]
```diff
@@ -0,0 +1,10 @@
+locals {
```
Comment (AlexBMes) [inline tests/gcp/unit_tests/goole_folder/locals.tf]: Should be a relative symlink. I.e.  `../../../../Google/resource-manager/folder/locals.yaml`

File: tests/gcp/unit_tests/goole_folder/resources/folders.yaml [hunk 10]
```diff
@@ -0,0 +1,11 @@
+project_id: "decisive-coda-355402"
```
Comment (AlexBMes) [inline tests/gcp/unit_tests/goole_folder/resources/folders.yaml]: Should use non existent example project and folder IDs

**Review by AlexBMes — CHANGES_REQUESTED** [review]
* The deployment tests should reference the folder module using source. This would involve deleting most of the files in the project and using:
  ```hcl
  module folders {
    source = "../../../../Google/resource-manager/folder/main.tf"
  }
  ```
  to reference to the folder resources. (see comments on files)

* Some of the files in the unit_tests directory should use relative symlinks instead of being a hard copy of the files. (see comments)

* Also if the `project_id` and `folder_id` values in the `folders.yaml` files are real projects and folders, might be better to replace with example id's that don't reference real projects/folders. And in deployment tests to have a value that can be substituted based on Github Secrets. (see comments)

File: Google/resource-manager/folder/folders.yaml (hunk 1 above)
Comment (dgavrischuk) [reply Google/resource-manager/folder/folders.yaml]: done

File: .github/workflows/deployment_tests.yml (hunk 2 above)
Comment (dgavrischuk) [reply .github/workflows/deployment_tests.yml:48]: done

File: tests/gcp/deployment/google_folder/locals.tf (hunk 3 above)
Comment (dgavrischuk) [reply tests/gcp/deployment/google_folder/locals.tf]: done

File: tests/gcp/deployment/google_folder/main.tf (hunk 4 above)
Comment (dgavrischuk) [reply tests/gcp/deployment/google_folder/main.tf]: done

File: tests/gcp/deployment/google_folder/provider.tf (hunk 5 above)
Comment (dgavrischuk) [reply tests/gcp/deployment/google_folder/provider.tf]: done

File: tests/gcp/deployment/google_folder/resource/folders.yaml (hunk 6 above)
Comment (dgavrischuk) [reply tests/gcp/deployment/google_folder/resource/folders.yaml]: done

File: tests/gcp/deployment/google_folder/resource/folders.yaml (hunk 7 above)
Comment (dgavrischuk) [reply tests/gcp/deployment/google_folder/resource/folders.yaml]: done

File: tests/gcp/unit_tests/goole_folder/locals.tf (hunk 9 above)
Comment (dgavrischuk) [reply tests/gcp/unit_tests/goole_folder/locals.tf]: done

File: tests/gcp/deployment/google_folder/vars.tf (hunk 8 above)
Comment (dgavrischuk) [reply tests/gcp/deployment/google_folder/vars.tf]: done

File: tests/gcp/unit_tests/goole_folder/resources/folders.yaml (hunk 10 above)
Comment (dgavrischuk) [reply tests/gcp/unit_tests/goole_folder/resources/folders.yaml]: done

**Review by AlexBMes — APPROVED** [review]

**Pull request closed without merging**